src/Elmish.WPF/BindingData.fs

```diff
@@ -134,30 +135,114 @@ and internal AlterMsgStreamData<'model, 'msg, 'bindingModel, 'bindingMsg, 'dispa
     |> this.AlterMsgStream
 
 
-and internal BaseBindingData<'model, 'msg> =
-  | OneWayData of OneWayData<'model, obj>
-  | OneWayToSourceData of OneWayToSourceData<'model, 'msg, obj>
-  | OneWaySeqData of OneWaySeqData<'model, obj, obj, obj>
-  | TwoWayData of TwoWayData<'model, 'msg, obj>
+and internal BaseBindingData<'model, 'msg, 't> =
+  | OneWayData of OneWayData<'model, 't>
+  | OneWayToSourceData of OneWayToSourceData<'model, 'msg, 't>
+  | OneWaySeqData of OneWaySeqData<'model, obj, 't, obj>
+  | TwoWayData of TwoWayData<'model, 'msg, 't>
   | CmdData of CmdData<'model, 'msg>
-  | SubModelData of SubModelData<'model, 'msg, obj, obj, obj>
-  | SubModelWinData of SubModelWinData<'model, 'msg, obj, obj, obj>
-  | SubModelSeqUnkeyedData of SubModelSeqUnkeyedData<'model, 'msg, obj, obj, obj, obj>
-  | SubModelSeqKeyedData of SubModelSeqKeyedData<'model, 'msg, obj, obj, obj, obj, obj>
+  | SubModelData of SubModelData<'model, 'msg, obj, obj, 't>
+  | SubModelWinData of SubModelWinData<'model, 'msg, obj, obj, 't>
+  | SubModelSeqUnkeyedData of SubModelSeqUnkeyedData<'model, 'msg, obj, obj, obj, 't>
+  | SubModelSeqKeyedData of SubModelSeqKeyedData<'model, 'msg, obj, obj, obj, 't, obj>
   | SubModelSelectedItemData of SubModelSelectedItemData<'model, 'msg, obj>
 
 
-and internal BindingData<'model, 'msg> =
-  | BaseBindingData of BaseBindingData<'model, 'msg>
-  | CachingData of BindingData<'model, 'msg>
-  | ValidationData of ValidationData<'model, 'msg>
-  | LazyData of LazyData<'model, 'msg, obj, obj>
-  | AlterMsgStreamData of AlterMsgStreamData<'model, 'msg, obj, obj, obj>
+and internal BindingData<'model, 'msg, 't> =
+  | BaseBindingData of BaseBindingData<'model, 'msg, 't>
+  | CachingData of BindingData<'model, 'msg, 't>
+  | ValidationData of ValidationData<'model, 'msg, 't>
+  | LazyData of LazyData<'model, 'msg, obj, obj, 't>
+  | AlterMsgStreamData of AlterMsgStreamData<'model, 'msg, obj, obj, obj, 't>
 
 
 
 module internal BindingData =
 
+  module private MapT =
+
+    let baseCase (fOut: 't0 -> 't1) (fIn: 't1 -> 't0) =
+      function
+      | OneWayData d -> OneWayData {
+          Get = d.Get >> fOut
+        }
+      | OneWayToSourceData d -> OneWayToSourceData {
+          Set = fIn >> d.Set
+        }
+      | OneWaySeqData d -> OneWaySeqData {
+          Get = d.Get
+          CreateCollection = d.CreateCollection >> CollectionTarget.mapCollection fOut
+          GetId = d.GetId
+          ItemEquals = d.ItemEquals
+        }
+      | TwoWayData d -> TwoWayData {
+          Get = d.Get >> fOut
+          Set = fIn >> d.Set
+        }
+      | CmdData d -> CmdData {
+          Exec = d.Exec
+          CanExec = d.CanExec
+          AutoRequery = d.AutoRequery
+        }
```

Comment:
```suggestion
      | CmdData d -> CmdData d
```

Reply:
This doesn't make the function infer less generically.

Reply:
I left it like that for consistency. Honestly I'd about rather have consistency than fewer lines in this case.